### PR TITLE
Add CRL checking for client certificate

### PR DESCRIPTION
**Pull request by Vagabond**
If the client provides a certificate, make sure it is actually valid AND
not revoked.

See also basho/riak_test#416

### CODE REVIEW

**Comment by Vagabond** [comment]
Oh, for this to actually work, you'll need to patch public_key:

https://github.com/erlang/otp/pull/97

File: src/riak_api_pb_server.erl [hunk 1]
```diff
@@ -520,4 +523,172 @@ send_error_and_flush(Error, State) ->
...
+    end.
+
+fetch([]) ->
+    not_available;
+fetch([{uniformResourceIdentifier, "file://"++_File}|Rest]) ->
+    lager:debug("fetching CRLs from file URIs is not supported"),
+    fetch(Rest);
+fetch([{uniformResourceIdentifier, "http"++_=URL}|Rest]) ->
+    lager:debug("getting CRL from ~p~n", [URL]),
+    inets:start(),
```
Comment (cmeiklejohn) [inline src/riak_api_pb_server.erl]: Do we have to worry about an error being returned here because it's already been started?

**Comment by cmeiklejohn** [comment]
I don't mean to be overly pedantic here, but adding some docstrings to the functions would go a long way to being able to revisit this code later and know why it does certain things.

File: src/riak_api_pb_server.erl [hunk 2]
```diff
@@ -520,4 +523,172 @@ send_error_and_flush(Error, State) ->
...
+                                                 end
+                                         end, [], CRLDistPoints),
+            public_key:pkix_crls_validate(Cert, DPointsAndCRLs, [{issuer_fun, {fun issuer_function/4, State}}])
+    end.
 
+issuer_function(_DP, CRL, {_, _Issuer}, {TrustedCAs, IntermediateCerts}) ->
+    %% XXX the 'Issuer' we get passed here is actually a lie, public key treats the Authority Key Identifier as the 'issuer'
+    %% Read the CA certs out of the file
+    %{ok, Bin} = file:read_file(CACerts),
+    %CACerts = load_certs("/home/andrew/lavabit"),
```
Comment (cmeiklejohn) [inline src/riak_api_pb_server.erl]: These commented lines probably shouldn't be here.

File: src/riak_api_pb_server.erl [hunk 3]
```diff
@@ -520,4 +523,172 @@ send_error_and_flush(Error, State) ->
...
+                  false ->
+                      undefined
+              end;
+         (_E, Acc) ->
+              %% already found a match
+              Acc
+      end, undefined, Certs).
+
+fetch_point(#'DistributionPoint'{distributionPoint = P}) ->
+    case P of
```
Comment (cmeiklejohn) [inline src/riak_api_pb_server.erl]: Is this really necessary?  Seems like you could just explicitly match P in the function clause?

**Comment by cmeiklejohn** [comment]
:+1: once #40 is merged.  Should address most of my comments in this PR.

**Pull request merged**